ec2/spark_ec2.py

```diff
@@ -1259,6 +1259,15 @@ def real_main():
             cluster_instances=(master_nodes + slave_nodes),
             cluster_state='ssh-ready'
         )
+        
+        # Determine types of running instances
+        existing_master_type = master_nodes[0].instance_type
+        existing_slave_type = slave_nodes[0].instance_type
+        if existing_master_type == existing_slave_type:
+            existing_master_type = ""
+        opts.master_instance_type = existing_master_type
```

Comment:
Does this not force the instance types to match the existing ones? that's not inherently required, right? I might misunderstand.